modules/dop/providers/projectpipeline/service_impl.go

```diff
@@ -1583,3 +1573,84 @@ func getRemotes(appNames []string, orgName, projectName string) []string {
 	}
 	return remotes
 }
+
+func (p *ProjectPipelineService) OneClickCreate(ctx context.Context, params *pb.OneClickCreateProjectPipelineRequest) (*pb.OneClickCreateProjectPipelineResponse, error) {
+	if err := params.Validate(); err != nil {
+		return nil, apierrors.ErrOneClickCreateProjectPipeline.InvalidParameter(err)
+	}
+	if len(params.PipelineYmls) == 0 {
+		return nil, apierrors.ErrOneClickCreateProjectPipeline.InvalidParameter(fmt.Errorf("the pipelineYmls is empty"))
+	}
+
+	// permission check
+	err := p.checkRolePermission(apistructs.IdentityInfo{
+		UserID: apis.GetUserID(ctx),
+	}, &apistructs.PipelineCreateRequestV2{
+		Labels: map[string]string{
+			apistructs.LabelAppID:  strconv.FormatUint(params.AppID, 10),
+			apistructs.LabelBranch: params.Ref,
+		},
+	}, apierrors.ErrOneClickCreateProjectPipeline)
+	if err != nil {
+		return nil, err
+	}
+
+	createReq := make([]*pb.CreateProjectPipelineRequest, 0, len(params.PipelineYmls))
```

Comment:
set a limitation like 10 yaml to create, exceed the limitation will not be created.

Reply:
done

Reply:
done